language/move-lang/src/typing/translate.rs

```diff
@@ -87,30 +87,41 @@ fn main_function_(context: &mut Context, name: FunctionName, f: N::Function) ->
 }
 
 fn check_primitive_main_arg(context: &mut Context, mloc: Loc, ty: &Type) {
+    use BuiltinTypeName_ as BT;
+
     let sp!(loc, ty_) = ty;
-    match (ty_.builtin_name(), ty_) {
-        (Some(_), Type_::Apply(_, _, ss)) => ss
-            .iter()
-            .for_each(|s| check_primitive_main_arg(context, mloc, s)),
-        _ => check_primitive_main_arg_error(
-            context,
-            mloc,
-            *loc,
-            core::error_format(ty, &Subst::empty()),
-        ),
+    if let Some(bt) = ty_.builtin_name() {
+        match bt.value {
+            BT::U8 | BT::U64 | BT::U128 | BT::Bool | BT::Address => return,
+            BT::Vector => {
+                let vector_u8_ty = Type::unsafe_no_loc(Type_::Apply(
+                    None,
+                    TypeName::unsafe_no_loc(TypeName_::Builtin(BuiltinTypeName::unsafe_no_loc(
+                        BuiltinTypeName_::Vector,
+                    ))),
+                    vec![Type::unsafe_no_loc(Type_::Apply(
+                        None,
+                        TypeName::unsafe_no_loc(TypeName_::Builtin(
+                            BuiltinTypeName::unsafe_no_loc(BuiltinTypeName_::U8),
+                        )),
+                        vec![],
+                    ))],
+                ));
```

Comment:
@tnowacki does this look good to you? I'm not sure if there's a better way to construct a vector<u8>.

Reply:
There are a lot of utility functions for this. 
Also, never use `unsafe_no_loc` :) 
```
let vector_u8_ty = Type_::vector(loc, Type_::u8(loc));
```

Reply:
Great! Exactly what I'm looking for.